gapis/gfxapi/vulkan/custom_replay.go

```diff
@@ -1279,25 +1284,30 @@ func (a *RecreateImage) Mutate(ctx log.Context, s *gfxapi.State, b *builder.Buil
 func (a *RecreateBuffer) Mutate(ctx log.Context, s *gfxapi.State, b *builder.Builder) error {
 	o := a.Extras().Observations()
 	o.ApplyReads(s.Memory[memory.ApplicationPool])
+	createInfo := a.PCreateInfo.Slice(0, 1, s).Index(0, s).Read(ctx, a, s, b)
```

Comment:
Seems like that will be wrong, I get this error message when I try to replay (use gapic):
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
  panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: ⦕runtime error: invalid memory address or nil pointer derefer...
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x18b3311]

goroutine 134 [running]:
panic(0x2713440, 0xc420c8f3b0)
	/usr/local/google/home/qining/Workspace/tools/go/src/runtime/panic.go:500 +0x1a1
github.com/google/gapid/gapis/resolve.(*ContextListResolvable).Resolve.func1(0x3eab260, 0xc420385670, 0xc420cd7d28, 0xc420cd7ec8)
	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/contexts.go:89 +0x43e
panic(0x246cd80, 0xc420010160)
	/usr/local/google/home/qining/Workspace/tools/go/src/runtime/panic.go:458 +0x243
github.com/google/gapid/gapis/gfxapi/vulkan.(*RecreateBindAndFillBufferMemory).Mutate(0xc4203aaa20, 0x3eab260, 0xc420385670, 0xc4207ae5b0, 0x0, 0x1, 0x0)
	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/gfxapi/vulkan/custom_replay.go:1308 +0x111
github.com/google/gapid/gapis/resolve.(*ContextListResolvable).Resolve(0xc4207440d0, 0x3eab260, 0xc420385670, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/resolve/contexts.go:96 +0x2fd
github.com/google/gapid/gapis/database.(*memory).resolve.func1(0x3e2c180, 0xc4207440d0, 0xc420c267d0, 0xc420627a60, 0x3eab260, 0xc420385590)
	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/memory.go:123 +0x4e
created by github.com/google/gapid/gapis/database.(*memory).resolve
	/usr/local/google/home/qining/Workspace/gopath/src/github.com/google/gapid/gapis/database/memory.go:133 +0x5fe
```